src/components/Search/index.tsx

```diff
@@ -159,9 +237,80 @@ const Search = ({ dispatch, hasBorder, content }: { dispatch: AppDispatch; hasBo
           }
         }}
       />
+
       {hasBorder && <SearchIconButton greenIcon />}
+
+      {searchSuggestions.length > 0 && searchValue ? (
+        <SuggestionsDropdown
+          suggestions={searchSuggestions}
+          onSuggestionSelect={onSuggestionSelect}
+          expandSuggestions={expandSuggestions}
+          setExpandSuggestions={setExpandSuggestions}
+        />
+      ) : null}
     </SearchPanel>
   )
 }
 
+interface SearchSuggestion {
+  value?: string
+  balance?: number
+  path?: string
+  type: string
+}
+
+const generateSuggestionValue = (suggestion: SearchSuggestion) => {
+  if (isMobile() && suggestion.type !== 'Block Height' && suggestion.value) {
+    return parseLongAddressHashMobile(suggestion.value)
+  }
+  return suggestion.value
+}
+
+const generateFormattedBalance = (suggestion: SearchSuggestion) => {
+  if (suggestion.balance) {
+    return `${(suggestion.balance / 10 ** 8).toFixed(3)} ckb`
+  }
+  return ''
+}
+
+const SuggestionsDropdown = ({
+  suggestions = [],
+  onSuggestionSelect,
+  expandSuggestions,
+  setExpandSuggestions,
+}: {
+  suggestions: SearchSuggestion[]
+  onSuggestionSelect: any
+  expandSuggestions: boolean
+  setExpandSuggestions: any
+}) => (
+  <SuggestionsPanel>
+    <SuggestionHeading>{suggestions.length > 0 && suggestions[0].type}</SuggestionHeading>
+    {suggestions
+      .slice(0, expandSuggestions ? suggestions.length : 3)
+      .filter(suggestion => suggestion.type !== 'Error')
+      .map(suggestion => {
+        return (
+          <SuggestionButton
+            key={`${suggestion.value}${suggestions.findIndex(s => s.value === suggestion.value)}`}
+            type="button"
+            onClick={() => onSuggestionSelect(suggestion)}
+          >
+            <SuggestionValue>{generateSuggestionValue(suggestion)}</SuggestionValue>
+            <SuggestionBalance>{generateFormattedBalance(suggestion)}</SuggestionBalance>
+          </SuggestionButton>
+        )
+      })}
+    {suggestions.length > 3 && !expandSuggestions && (
+      <SuggestionExpand
+        onClick={() => {
+          setExpandSuggestions(true)
+        }}
+      >
+        More
+      </SuggestionExpand>
+    )}
+  </SuggestionsPanel>
+)
+
```

Comment:
When searching fail, suggestion dropdown should display `Searching fail !` not `Error`.